src/rgblob/game.cljs

```diff
@@ -0,0 +1,218 @@
+                  :green {:diff 10 :same [11 14]}
+                  :blue  {:diff 15 :same [16 19]}})
+
+(def tile-anim [[:white-diff [0 0]] [:white-same [0 0]]
+                [:black-diff [1 1]] [:black-same [1 1]]
+                [:red-diff [5 5]] [:red-same [6 9]]
+                [:green-diff [10 10]] [:green-same [11 14]]
+                [:blue-diff [15 15]] [:blue-same [16 19]]])
+
+(defn setup-tile-color-anims [sprite]
+    (doall (map (fn [[kw range]]
+        (log kw range)
+        (rs/add-anim! sprite kw range)) tile-anim)))
+
+(defn get-color-at-board-pos [x y]
+    (nth (nth board y) x))
+
+(defn color->board-anim [color t]
+  (keyword (str (name color) "-" (name t))))
+
+(defn build-board-sprite-row [row-number]
+  (map  (fn [x] (let [sprite (rs/sprite { 
+                                   :src "/img/tiles.png"
+                                   :x (* x 64) :y (* row-number 64)
+                                   :width 160
+                                   :height 128
+                                   :frame-width 32
+                                   :frame-height 32
+                                   :scale-x 2
+                                   :scale-y 2})]
+                 (setup-tile-color-anims sprite)
+                 (let [board-anim (color->board-anim 
+                                    (get-color-at-board-pos x row-number) 
+                                    :diff)] 
+                   (log board-anim)
+                   (rs/play! sprite board-anim))
+                 sprite))
+           (range 8)))
+
+(defonce board-sprites
+    (map build-board-sprite-row (range 8)))
+
+(defn input 
+    "Puts a user's intended input into the state for calculation"
+    [intent]
+    (player/input! intent))
+
+(defn step-all! [sprites]
+    ; (log "stepping all of them" sprites)
+  (dorun (map #(rs/step! %) sprites)))
+
+(defn get-context []
+    (let [id (:canvas-id @state)
+          canvas (.getElementById js/document id)]
+      (.getContext canvas "2d")))
+
+(defn update-tile-frame [sprite color pcolor]
+    (let [t (if (= color pcolor) :same :diff)]
+        (swap! sprite assoc :frame (get-in tile-frames [color t]))))
+
+(defn get-flat-board-sprites []
+    (apply concat board-sprites))
+
+(defn get-coords-from-sprite [sprite]
+    (let [{x :x y :y 
+           fw :frame-width fh :frame-height 
+           sx :scale-x sy :scale-y} @sprite
+          rw (* fw sx)
+          rh (* fh sy)
+          cx (if (> x 0) (Math/floor (/ x rw)) 0)
+          cy (if (> y 0) (Math/floor (/ y rh)) 0)]
+         {:x cx :y cy}))
+
+(defn render-board-row [row]
+    ; (log "row" row)
+    (let [player-color (player/get-color)]
+      (reduce 
+          (fn [_ sprite]
+            (rs/clear (get-context) sprite)
+            (rs/render (get-context) sprite )) 
+          true row)))
+
+(defn render-board []
+    (reduce (fn [_ row] #_(log-obj row) (render-board-row row)) true board-sprites))
```

Comment:
I wouldn't leave `#_(log-obj row)` in here.  If you're not using it, probably best to pull it out.